riscv/mmu.cc

```diff
@@ -161,6 +161,34 @@ void mmu_t::load_slow_path(reg_t addr, reg_t len, uint8_t* bytes, uint32_t xlate
   }
 }
 
+void mmu_t::clean_inval(reg_t addr, bool clean, bool inval)
+{
+  reg_t paddr = addr & ~(blocksz - 1);
+  try {
+    paddr = translate(paddr, blocksz, LOAD, 0);
+  } catch (trap_load_access_fault& t) {
+    trap_store_access_fault(t.has_gva(), addr, 0, 0);
+  } catch (trap_load_page_fault& t) {
+    trap_store_page_fault(t.has_gva(), addr, 0, 0);
+  } catch (trap_load_guest_page_fault& t) {
+    trap_store_guest_page_fault(addr, t.get_tval2(), 0);
+  } catch (...) {
+    abort();
+  }
+  if (auto host_addr = sim->addr_to_mem(paddr)) {
+    if (tracer.interested_in_range(paddr, paddr + PGSIZE, LOAD))
+      tracer.clean_invalidate(paddr, blocksz, clean, inval);
+  } else {
+    throw trap_store_access_fault((proc) ? proc->state.v : false, addr, 0, 0);
+  }
+}
+
+void mmu_t::cbo_zero(reg_t addr) {
+  auto base = addr & ~(blocksz - 1);
+  for (size_t offset = 0; offset < blocksz; offset += 4)
```

Comment:
I recall suggesting incrementing by 1, to capture the architectural intent, but I also vaguely recall someone stating that incrementing by 4 was sufficient?

Reply:
OK. I'll update this.